src/client/batch.rs

```diff
@@ -1,191 +1,106 @@
-use serde::Deserialize;
-
-use std::fs::File;
-
-use super::*;
-
+use csv;
 use rust_decimal::Decimal;
-use rust_decimal_macros::dec;
+use std::path::Path;
+use std::str::FromStr;
 
-#[derive(Debug, Deserialize)]
-pub struct PaymentInput {
-    pub username: String,
-    pub usd: Decimal,
-    pub memo: Option<String>,
-}
+use crate::client::GaloyClient;
+use crate::client::Wallet;
 
-impl From<PaymentInput> for Payment {
-    fn from(input: PaymentInput) -> Payment {
-        Payment {
-            username: input.username,
-            usd: input.usd,
-            sats: None,
-            wallet_id: None,
-            memo: input.memo,
-        }
+// Utility function to check if file exists
+pub fn check_file_exists(file: &str) -> anyhow::Result<()> {
+    let file_path = Path::new(&file);
+    if !file_path.exists() {
+        return Err(anyhow::anyhow!("File not found: {}", file));
     }
+    Ok(())
 }
 
-#[derive(Debug)]
-struct Payment {
-    username: String,
-    usd: Decimal,
-    sats: Option<Decimal>,
-    wallet_id: Option<String>,
-    memo: Option<String>,
-}
-
-pub struct Batch {
-    payments: Vec<Payment>,
-    client: GaloyClient,
-    /// price in btc/usd
-    price: Decimal,
-}
-
-impl Batch {
-    pub fn new(client: GaloyClient, price: Decimal) -> Self {
-        let payments: Vec<Payment> = vec![];
-        Self {
-            payments,
-            client,
-            price,
-        }
+// Utility function to read and validate the CSV file
```

Comment:
This comment is redundant as it adds no additional information than the name of the function. Delete it - or alternatively if you want to publish this as a library turn in into a doc style comment (`///`) and add something useful.